### PR TITLE
Trev8939/map view dispose

### PR DESCRIPTION
**Pull request by TADraeseke**
This PR covers the following code fixes:
- Remove `android:configChanges="..."` from all modules EXCEPT 'Feature layer rendering mode map' and 'Feature layer rendering mode scene' because of hard crash on device rotation during zoom events
- Adds `MapView/SceneView.dispose()`to the `onDestroy()` callback in all modules with a `SceneView/MapView`
- Unifies all samples to ESRI preferred code formatting (ie two spaces instead of tab, enums or inner classes at bottom, etc)
- Makes all TAGs derived from `= CLASSNAME.class.getSimpleName();`

During the course of this PR all modules were tested for rotation handling and for general functionality.

### CODE REVIEW

File: java/create-geometries/src/main/java/com/esri/arcgisruntime/sample/creategeometries/MainActivity.java [hunk 1]
```diff
@@ -43,6 +43,40 @@
  */
 public class MainActivity extends AppCompatActivity {
 
+  private MapView mMapView;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
```
Comment (shellygill) [inline java/create-geometries/src/main/java/com/esri/arcgisruntime/sample/creategeometries/MainActivity.java]: Why was the order of methods changed here? This order was deliberate, to show the creation of geometries as the first code in the sample (code guidelines indicate constructors should be before class methods, but this isn't a constructor).

File: java/feature-layer-feature-service/src/main/java/com/esri/arcgisruntime/samples/featurelayerfeatureservice/MainActivity.java [hunk 2]
```diff
@@ -27,73 +25,54 @@
...
-    public boolean onCreateOptionsMenu(Menu menu) {
-        // Inflate the menu; this adds items to the action bar if it is present.
-        getMenuInflater().inflate(R.menu.menu_main, menu);
-        return true;
-    }
+    // add the layer to the map
+    map.getOperationalLayers().add(featureLayer);
 
-    @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
```
Comment (shellygill) [inline java/feature-layer-feature-service/src/main/java/com/esri/arcgisruntime/samples/featurelayerfeatureservice/MainActivity.java]: Getting rid of the options menu is a good idea as it's unused - but should delete the menu_main resource file too in this case.

**Comment by shellygill** [comment]
@TADraeseke a few things to address in addition to the couple of inline-comments:
- Kotlin samples need similar changes?
- `android:configChanges="orientation|screenSize"` still appears in feature-layer-rendering-mode and feature-layer-rendering-mode-scene
- open-existing-map sample has code that deals with config changes in the MainActivity, and also lacks the `dispose` call.
- viewshed-geoprocessing lacks the `dispose` call

File: java/colormap-renderer/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <resources>
     <string name="app_name">Colormap Renderer</string>
     <string name="location_permission_denied">Colormap Renderer could not open the raster because WRITE permission was denied.</string>
-    <string name="config_data_sdcard_offline_dir">/ArcGIS/samples/raster</string>
+    <string name="config_data_sdcard_offline_dir">ArcGIS/samples/raster</string>
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply java/colormap-renderer/src/main/res/values/strings.xml]: yep! Was printing out a path with double /

**Comment by shellygill** [comment]
@TADraeseke - build finally finished and shows one more problem - extra close curly brackets in web-tiled-layer sample MainActivity line 88

File: java/create-geometries/src/main/java/com/esri/arcgisruntime/sample/creategeometries/MainActivity.java (hunk 1 above)
Comment (TADraeseke) [reply java/create-geometries/src/main/java/com/esri/arcgisruntime/sample/creategeometries/MainActivity.java]: oops, ok. Reverted!

File: java/feature-layer-feature-service/src/main/java/com/esri/arcgisruntime/samples/featurelayerfeatureservice/MainActivity.java (hunk 2 above)
Comment (TADraeseke) [reply java/feature-layer-feature-service/src/main/java/com/esri/arcgisruntime/samples/featurelayerfeatureservice/MainActivity.java]: done

**Comment by TADraeseke** [comment]
Thanks @shellygill - Two follow up questions.

1) Left the configChanges in in feature-layer-rendering-mode and feature-layer-rendering-mode-scene to avoid a hard crash if the device is rotated during the 'zoom animation' of the samples. Didn't know if this is a good reason to leave it in or not.

2) Added `dispose` call to open-existing-map, but is it ok if the sample has code in it to handle config changes? Or is that something I should remove? I'd have to dig into the code a bit to see what needs to go if it should be removed.

**Comment by TADraeseke** [comment]
@shellygill - configChanges removed from 'rendering mode samples' and removed onConfigurationChange code from 'open existing map'

I believe thats all changes made!

File: kotlin/display-map/src/main/java/com/esri/arcgisruntime/sample/displaymap/MainActivity.kt [hunk 4]
```diff
@@ -16,45 +16,50 @@
 
 package com.esri.arcgisruntime.sample.displaymap
 
-import android.support.v7.app.AppCompatActivity
-import android.os.Bundle
-
 //[DocRef: Name=Import map types-Android, Category=Get started, Topic=Develop your first map app with Kotlin]
-import com.esri.arcgisruntime.mapping.ArcGISMap
-import com.esri.arcgisruntime.mapping.Basemap
 //[DocRef: END]
```
Comment (shellygill) [inline kotlin/display-map/src/main/java/com/esri/arcgisruntime/sample/displaymap/MainActivity.kt]: This has broken the doc snippet.

File: kotlin/display-map/src/main/java/com/esri/arcgisruntime/sample/displaymap/MainActivity.kt [hunk 5]
```diff
@@ -16,45 +16,50 @@
...
 
 //[DocRef: Name=Import kotlinx-Android, Category=Get started, Topic=Develop your first map app with Kotlin]
```
Comment (shellygill) [inline kotlin/display-map/src/main/java/com/esri/arcgisruntime/sample/displaymap/MainActivity.kt]: Broken doc snippet.

**Comment by TADraeseke** [comment]
Thanks @shellygill - Should be sorted now!

**Pull request merged**